javascript/ql/src/semmle/javascript/frameworks/ServerLess.qll

```diff
@@ -61,7 +63,9 @@ private module ServerLess {
   private predicate hasServerlessHandler(File file, string func) {
     exists(File ymlFile, string handler, string codeURI, string fileName |
       hasServerlessHandler(ymlFile, handler, codeURI) and
+      // Captures everything right of the dot in `handler`. E.g. if `handler` is "index.foo" then `func` is "foo".
       func = handler.regexpCapture(".*\\.(.*)", 1) and
+      // Captures everything left of the dot in `handler`. E.g. if `handler` is "index.foo" then `fileName` is "index".
       fileName = handler.regexpCapture("([^.]+).*", 1)
```

Comment:
Could we unify the two regexps here? Adding another capture group to the first pattern should suffice.

I am not sure if the first capture group should be greedy, but I have just reused the current greediness. Intuitively, I would expect it to be lazy such that `index.foo.bar` would produce `index` and `foo.bar`.

```ql
exists(string pattern | pattern = "(.*)\\.(.*)" | 
  func = handler.regexpCapture(pattern, 1) and 
  fileName = handler.regexpCapture(pattern, 2)
)
```